code/game/objects/items.dm

```diff
@@ -1092,4 +1092,69 @@ var/global/list/image/blood_overlays = list()
 	if(shrapnel_type)
 		return new shrapnel_type(src)
 	else
-		return 0
\ No newline at end of file
+		return 0
+
+/obj/item/duplicate(atom/destination)
+	var/obj/item/dupe = ..()
+
+	if(iscarbon(loc) && iscarbon(destination))
```

Comment:
The basic parent `duplicate()` proc duplicates contents, since that's relatively safe.
Since clothing worn by humans and monkeys is placed in the human's contents, duplicating a human will duplicate all the human's clothes and place them in the duplicate's contents.

For the purpose of assigning each piece of clothing to the correct inventory slots, putting this code in `human/duplicate()` wouldn't work because the code would have no way to guarantee which original item each duplicate item came from, and so no way to confirm which inventory slot the duplicate should be placed in.